pkg/kubelet/cm/helpers_linux_test.go

```diff
@@ -197,3 +199,85 @@ func TestMilliCPUToQuota(t *testing.T) {
 		}
 	}
 }
+
+func TestHugePageLimits(t *testing.T) {
+	Mi := int64(1024 * 1024)
+	type inputStruct struct {
+		key   string
+		input string
+	}
+
+	testCases := []struct {
+		name     string
+		inputs   []inputStruct
+		expected map[int64]int64
+	}{
+		{
+			name: "no valid hugepages",
+			inputs: []inputStruct{
+				{
+					key:   "2Mi",
+					input: "128",
+				},
+			},
+			expected: map[int64]int64{},
+		},
+		{
+			name: "2Mi only",
+			inputs: []inputStruct{
+				{
+					key:   v1.ResourceHugePagesPrefix + "2Mi",
+					input: "128",
+				},
+			},
+			expected: map[int64]int64{2 * Mi: 128},
+		},
+		{
+			name: "2Mi and 4Mi",
+			inputs: []inputStruct{
+				{
+					key:   v1.ResourceHugePagesPrefix + "2Mi",
+					input: "128",
+				},
+				{
+					key:   v1.ResourceHugePagesPrefix + strconv.FormatInt(2*Mi, 10),
+					input: "256",
+				},
+				{
+					key:   v1.ResourceHugePagesPrefix + "4Mi",
+					input: "512",
+				},
+				{
+					key:   "4Mi",
+					input: "1024",
+				},
+			},
+			expected: map[int64]int64{2 * Mi: 384, 4 * Mi: 512},
+		},
+	}
+
+	for _, testcase := range testCases {
+		t.Run(fmt.Sprintf("%s", testcase.name), func(t *testing.T) {
+			resourceList := v1.ResourceList{}
+
+			for _, input := range testcase.inputs {
+				value, err := resource.ParseQuantity(input.input)
+				if err == nil {
+					resourceList[v1.ResourceName(input.key)] = value
+				}
+			}
+
+			resultValue := HugePageLimits(resourceList)
+
+			if len(testcase.expected) != len(resultValue) {
+				t.Errorf("unexpected length in result, expected: %d, actual: %d", len(testcase.expected), len(resultValue))
+			}
+			for k, v := range testcase.expected {
```

Comment:
why not use reflect.DeepEqual and also print the full maps when something doesn't match?